source/sessions/driver-sessions.rst

```diff
@@ -1197,6 +1212,25 @@ Instead, we require users to pass session as a parameter to each function::
   collection.insertOne(session, {})
   session.endSession()
 
+Why does a network error cause the ``ServerSession`` to be discarded from the pool?
+-----------------------------------------------------------------------------------
+
+When a network error is encountered when executing an operation with a
+``ClientSession``, the operation may be left running on the server. Re-using
+this ``ServerSession`` can lead to parallel operations which violates the
+rule that a session must be used sequentially. This results in multiple
+problems:
+
+#. killSessions to end an earlier operation would surprisingly also end a
+   later operation.
+#. An otherwise unrelated operation that just happens to use that same server
+   session will potentially block waiting for the previous operation to
+   complete. For example, a transactional write will block a subsequent
+   transactional write.
```

Comment:
Am I correct in assuming that the server doesn't prohibit a session from being attached to two (or more) concurrent operations? Or does such a prohibition depend on whether we're operating within a transaction?

Outside of the context of a network error, is it possible that drivers could already spawn parallel operations on the server using the same session? One scenario that comes to mind would be if a driver actually executed write commands from an unordered bulk write in parallel (either in or outside of a transaction), and all ensuing write commands use the same LSID.

Reply:
When both operations on the same session are part of a transaction (retryable writes/transactions), the server does not allow concurrent execution; the operations will be executed serially. 

When one (or both) operations on the same session are not transactional, the server allows concurrent execution.

> Outside of the context of a network error, is it possible that drivers could already spawn parallel operations on the server using the same session?

I am not aware of any other cases.

>  One scenario that comes to mind would be if a driver actually executed write commands from an unordered bulk write in parallel (either in or outside of a transaction), and all ensuing write commands use the same LSID.

This would be a bug in the driver and retryable writes would not work correctly.